plottr/plot/mpl.py

```diff
@@ -724,6 +742,54 @@ def setAllowedPlotTypes(self, *args: PlotType) -> None:
 
         self._currentlyAllowedPlotTypes = args
 
+    def selectComplexType(self, comp: ComplexRepresentation) -> None:
+        """makes sure that the selected `comp` is active (checked), all
+        others are not active.
+
+        This method should be used to catch a trigger from the UI.
+
+        If the active plot type has been changed by using this method,
+        we emit `complexPolarSelected`.
+        """
+        # deselect all other types
+        for k, v in self.ComplexActions.items():
+            if k is not comp and v is not None:
+                v.setChecked(False)
+
+        # don't want un-toggling - can only be done by selecting another type
+        self.ComplexActions[comp].setChecked(True)
+
+        if comp is not self._currentPlotType:
+            self._currentComplex = comp
+            self.complexPolarSelected.emit(self._currentComplex)
+
+    def setAllowedComplexTypes(self, *args: ComplexRepresentation) -> None:
```

Comment:
Perhaps we can come up with a better name than `args` `complex_representations` perhaps.